### PR TITLE
fix(CLOUDDST-23731): cosign after push

### PR DESCRIPTION
**Pull request by emilyzheng**
Cosign sign requires images already been pushed to destination repo. The commit also fixed an environment variable in cosign task.

### CODE REVIEW

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml [hunk 1]
```diff
@@ -25,11 +25,11 @@ spec:
     - name: sign-image
       image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
       env:
-        - name: AWS_REGION
+        - name: AWS_DEFAULT_REGION
```
Comment (mmalina) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:28]: Why this change? Was that a mistake?

Comment (mmalina) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:28]: Oh, you already mention it in the commit message, my bad. (But not in the PR description, so it could be added there as well.

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml
Comment (mmalina) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: I'm wondering if it was feasible actually sign before the push. I guess in theory you could sign the origin (the image in quay.io/redhat-user-workloads) instead of the target and then the `cosign copy` in `push-snapshot` would probably copy the signature as well? But the main reason rh-sign-image was moved before push-snapshot is that it's less reliable and if it fails, we don't want to end up with unsigned pushed images. I guess here the chance of it failing is lower. So I'm not really suggesting to change anything, just thinking out loud.

Comment (emilyzheng) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: Good point. Pub has to do it in the order of radas sign->push->cosign sign because it doesn't write to origin and it uses skopeo copy. But signing before push should be doable here as long as the task can push to origin. I will try it out.

Comment (mmalina) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: I am not sure if the docker secret we're using will have the permission to write to the redhat-user-workloads org either. To be honest I'm really not sure if it's a good idea. Maybe it would be better to wait for Ralph or Scott to ask their opinion. (And leave it as it is for now.)

**Review by mmalina — APPROVED** [review]

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml (hunk 1 above)
Comment (emilyzheng) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:28]: Added it to description.

Comment (mmalina) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:28]: Thanks.

**Comment by mmalina** [comment]
@emilyzheng can you sign the commit please?

**Comment by emilyzheng** [comment]
> @emilyzheng can you sign the commit please?

Signed now. I'd better avoid rebase from github UI next time :)

**Comment by mmalina** [comment]
> > @emilyzheng can you sign the commit please?
> 
> Signed now. I'd better avoid rebase from github UI next time :)

Cool, thanks. Yeah, the rebase via UI is usually not great :)
